bftengine/include/bftengine/KeyExchangeManager.hpp

```diff
@@ -31,27 +32,34 @@ class KeyExchangeManager {
  public:
   void exchangeTlsKeys(const SeqNum& bft_sn);
   // Generates and publish key to consensus
+  // TODO: persist the candidate
   void generateConsensusKeyAndSendInternalClientMsg(const SeqNum& sn);
   // Send the current main public key of the replica to consensus
   void sendMainPublicKey();
-  // Generates and publish the first replica's key,
-  void sendInitialKey(const ReplicaImp* repImpInstance, const SeqNum& = 0);
-  // The execution handler implementation that is called when a key exchange msg has passed consensus.
+
+  void waitForQuorum(const ReplicaImp* repImpInstance);
+
+  // Waits for a quorum and calls generateConsensusKeyAndSendInternalClientMsg
+  void waitForQuorumAndTriggerConsensusExchange(const ReplicaImp* repImpInstance, const SeqNum s = 0);
+  // The execution handler implementation that is called after a key exchange msg passed consensus.
+  // The new key pair will be used from two checkpoints after kemsg.generated_sn
   std::string onKeyExchange(const KeyExchangeMsg& kemsg, const SeqNum& req_sn, const std::string& cid);
```

Comment:
I think keyRotatation is the correct term

Reply:
Renaming will create inconsistencies with messages (KeyExchangeMsg)